### DISCUSSION TITLE
使用代理时aiohttp.client_exceptions.ClientHttpProxyError异常未捕获, 造成程序退出. 修改建议.

### DISCUSSION THREAD

**Discussion by thinkingl** [yutto-dev/yutto#89] [General]
如题, 未捕获代理异常, 造成程序直接退出, 没有重试.
我简单走读了一下代码发现问题应该是在  yutto/utils/fetcher.py:36 附近, except异常捕获中增加代理错误.

我本地修改如下:

                try:
                    return await connect_once(*args, **kwargs)
                except (
                    aiohttp.client_exceptions.ClientPayloadError,  # type: ignore
                    aiohttp.client_exceptions.ClientConnectorError,  # type: ignore
                    aiohttp.client_exceptions.ServerDisconnectedError,  # type: ignore
                    aiohttp.client_exceptions.ClientHttpProxyError, # type: ignore   ## <- 添加这个异常.
                ):

**Comment by SigureMo** [comment]
感谢，我加一下这个异常处理，稍后我发布一个新版本（可能是本周三或者更晚，在等 poetry 1.2.0 发布）
